src/context/WalletProvider/NewWalletViews/wallets/mipd/CoinbaseQrBody.tsx

```diff
@@ -0,0 +1,130 @@
+import { Alert, AlertDescription, AlertIcon, Button, Flex, Spinner } from '@chakra-ui/react'
+import { useCallback } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { CoinbaseIcon } from 'components/Icons/CoinbaseIcon'
+import { Text } from 'components/Text'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { CoinbaseConfig } from 'context/WalletProvider/Coinbase/config'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { useLocalWallet } from 'context/WalletProvider/local-wallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+const spinner = <Spinner color='white' />
+
+type CoinbaseQrBodyProps = {
+  isLoading: boolean
+  error: string | null
+  setIsLoading: (loading: boolean) => void
+  setError: (error: string | null) => void
+}
+
+export const CoinbaseQrBody = ({
+  isLoading,
+  error,
+  setIsLoading,
+  setError,
+}: CoinbaseQrBodyProps) => {
+  const translate = useTranslate()
+
+  const { dispatch, getAdapter } = useWallet()
+  const localWallet = useLocalWallet()
+
+  const pairDevice = useCallback(async () => {
+    setError(null)
+    setIsLoading(true)
+
+    const adapter = await getAdapter(KeyManager.Coinbase)
+    if (!adapter) {
+      setIsLoading(false)
+      return
+    }
+
+    try {
+      const wallet = await adapter.pairDevice()
+      if (!wallet) {
+        setError('walletProvider.errors.walletNotFound')
+        throw new Error(
+          `Call to hdwallet-${KeyManager.Coinbase.toLowerCase()}::pairDevice returned null or undefined`,
```

Comment:
q: it feels weird to have raw english in the app, it this error shown because caught upper in the tree?